lib/msf/core/post/windows/priv.rb

```diff
@@ -50,7 +51,11 @@ def initialize(info = {})
   def is_admin?
     if session_has_ext
       # Assume true if the OS doesn't expose this (Windows 2000)
-      return session.railgun.shell32.IsUserAnAdmin()['return'] rescue true
+      begin
+        return session.railgun.shell32.IsUserAnAdmin()['return']
+      rescue StandardError
+        true
```

Comment:
This might be out of scope since it was the previous behavior, but I'm wondering if there is a way to distinguish between an error in case the API is not exposed and a Railgun error due to something else (communication failure, process died, etc.). This might not be possible, but I'm worried about always assuming the session is an admin session if something goes wrong.

Reply:
Yeah, I'm a bit concerned about how much this PR already contains - happy to look at this afterwards.

Reply:
Although thinking about this a bit more, the one case that this is described as being applicable (Windows 2000) doesn't support Meterpreter anymore anyway. It feels like the exception handling here is a bit misguided - it's inferring from the presence of an exception that it's on Windows 2000, and returning true, perhaps because Windows 2000 just treated everyone as an admin? I'd be inclined just to remove that exception handling, given the documented reason for it existing no longer applies.

Reply:
>  It feels like the exception handling here is a bit misguided - it's inferring from the presence of an exception that it's on Windows 2000

The intention of the `rescue` here is to presume the user is an admin if `shell32.IsUserAnAdmin()` fails. Windows 2000 is not inferred.

This code is in the `is_admin?` method which is used as a privilege check to bail out prior to attempting operations on the host which are doomed to failure. Historically, in instances where the check failed, it was expected to fail safe so as to not prevent further execution, hence the greedy rescue.

One potential reason that `IsUserAnAdmin` may fail is due to using an old operating system such as Windows 2000.

Granted, the code is sloppy. If we wanted to always fail safe, we should rescue from method.